extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -149,16 +150,58 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
       return;
     }
 
-    this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
     this.onRtlStateUpdate_(this.storeService_.get(StateProperty.RTL_STATE));
-    this.mutateElement(() => {
-      createShadowRootWithStyle(
-        this.element,
-        this.shoppingTagEl_,
-        shoppingTagCSS
-      );
-      this.hasAppendedInnerShoppingTagEl_ = true;
-    });
+
+    this.measureMutateElement(
+      () => {
+        this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
+      },
+      () => {
+        createShadowRootWithStyle(
+          this.element,
+          this.shoppingTagEl_,
+          shoppingTagCSS
+        );
+        this.hasAppendedInnerShoppingTagEl_ = true;
+
+        const rightEdgeOfScreentoTagDistance =
+          document.getElementsByTagName('amp-story-shopping-attachment')[0]
+            ./*OK*/ clientWidth *
+            0.5 -
+          this.element./*OK*/ offsetLeft -
+          this.element./*OK*/ offsetWidth * 0.5;
+
+        const shoppingTagInner = this.element.shadowRoot.querySelector(
+          '.amp-story-shopping-tag-inner'
+        );
+
+        if (this.shoppingTagEl_.getAttribute('dir') !== 'rtl') {
+          if (rightEdgeOfScreentoTagDistance > -160) {
+            shoppingTagInner?.classList.remove('amp-story-shopping-tag-inner');
```

Comment:
Should this have the `i-amphtml-` prefix?

Reply:
The class in the .css file is .amp-story-shopping-tag-inner, so I'm not sure I need the prefix.

Reply:
Why doesn't `amp-story-shopping-tag-inner` have the prefix? They all should have it

Reply:
I added prefixes to the css classes and they will be pushed in the next commit.